solidity/contracts/liquidity-protection/LiquidityProtectionStore.sol

```diff
@@ -251,112 +287,94 @@ contract LiquidityProtectionStore is ILiquidityProtectionStore, Owned, TokenHand
+        uint256 id = nextPositionId;
+        nextPositionId += 1;
 
-        protectedLiquidities[id] = ProtectedLiquidity({
+        positions[id] = Position({
             provider: _provider,
             index: ids.length,
             poolToken: _poolToken,
             reserveToken: _reserveToken,
-            poolAmount: _poolAmount,
-            reserveAmount: _reserveAmount,
-            reserveRateN: _reserveRateN,
-            reserveRateD: _reserveRateD,
-            timestamp: _timestamp
+            poolAmount: toUint128(_poolAmount),
+            reserveAmount: toUint128(_reserveAmount),
+            reserveRateInfo: encodeReserveRateInfo(_reserveRateN, _reserveRateD, _timestamp)
         });
 
         ids.push(id);
 
-        // update the total amounts
-        totalProtectedPoolAmounts[_poolToken] = totalProtectedPoolAmounts[_poolToken].add(_poolAmount);
-        totalProtectedReserveAmounts[_poolToken][_reserveToken] = totalProtectedReserveAmounts[_poolToken][_reserveToken]
-            .add(_reserveAmount);
-
-        emit ProtectionAdded(_provider, _poolToken, _reserveToken, _poolAmount, _reserveAmount);
+        emit PositionAdded(id, _provider, _poolToken, _reserveToken, _poolAmount, _reserveAmount);
         return id;
     }
 
     /**
-     * @dev updates an existing protected liquidity pool/reserve amounts
+     * @dev updates an existing position pool/reserve amounts
      * can only be called by the contract owner
      *
-     * @param _id                  protected liquidity id
+     * @param _id                  position id
      * @param _newPoolAmount       new pool tokens amount
      * @param _newReserveAmount    new reserve tokens amount
      */
-    function updateProtectedLiquidityAmounts(
+    function updatePositionAmounts(
         uint256 _id,
         uint256 _newPoolAmount,
         uint256 _newReserveAmount
     ) external override ownerOnly greaterThanZero(_newPoolAmount) greaterThanZero(_newReserveAmount) {
-        // update the protected liquidity
-        ProtectedLiquidity storage liquidity = protectedLiquidities[_id];
+        // update the position
+        Position storage pos = positions[_id];
 
         // validate input
-        address provider = liquidity.provider;
-        require(provider != address(0), "ERR_INVALID_ID");
+        require(pos.provider != address(0), "ERR_INVALID_ID");
 
-        IDSToken poolToken = liquidity.poolToken;
-        IERC20Token reserveToken = liquidity.reserveToken;
-        uint256 prevPoolAmount = liquidity.poolAmount;
-        uint256 prevReserveAmount = liquidity.reserveAmount;
-        liquidity.poolAmount = _newPoolAmount;
-        liquidity.reserveAmount = _newReserveAmount;
+        IDSToken poolToken = pos.poolToken;
+        IERC20Token reserveToken = pos.reserveToken;
+        uint256 prevPoolAmount = uint256(pos.poolAmount);
+        uint256 prevReserveAmount = uint256(pos.reserveAmount);
+        pos.poolAmount = toUint128(_newPoolAmount);
```

Comment:
if we only support uint128, why not pass in uint128 as the arg types?

Reply:
@yudilevi : The original intention (before it was decided to redeploy the main contract), was to avoid breaking the API of the store contract.